tests/test_find_util.py

```diff
@@ -22,69 +22,106 @@
 DEALINGS IN THE SOFTWARE.
 """
 
+from __future__ import annotations
+
+from collections.abc import Callable, Iterable, Iterator
+from typing import Any
+
 import pytest
 from discord.utils import find
 
 
-def is_even(x):
+def is_even(x: int) -> bool:
     return x % 2 == 0
 
 
+def always_true(_: object) -> bool:
+    return True
+
+
+def greater_than_3(x: int) -> bool:
+    return x > 3
+
+
+def equals_1(x: int) -> bool:
```

Comment:
All of the equals_... should return `TypeIs[Literal[what they return]]`. Import `TypeIs` from `typing_extensions`.